test/issues/issues-test.js

```diff
@@ -533,4 +533,16 @@ describe("issues", function () {
             assert.same(calledWith, false);
         });
     });
+
+    describe("#1882", function () {
+        it("should use constructor name when checking deepEquality", function () {
+            class ClassWithoutProps {}
+            const arg1 = new ClassWithoutProps(); //arg1.constructor.name === ClassWithoutProps
+            const arg2 = new Proxy({}, {}); //arg2.constructor.name === Object
```

Comment:
Can this test be done without using [`Proxy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy), which is from ES6 and is not supported in IE11?

Perhaps create `AnotherClassWithoutProps`?

Reply:
That's true I will rewrite with to your suggestion, which makes the most sense I think